web/packages/teleterm/src/ui/DocumentCluster/DocumentCluster.tsx

```diff
@@ -52,97 +53,144 @@ export default function DocumentCluster(props: {
       });
   }, [appCtx.workspacesService, clusterName, clusterUri, props.doc.uri]);
 
-  function logIn(): void {
-    appCtx.modalsService.openRegularDialog({
-      kind: 'cluster-connect',
-      clusterUri,
-      reason: undefined,
-      prefill: undefined,
-      onCancel: () => {},
-      onSuccess: () => {},
-    });
-  }
+  const [clusterSyncAttempt, syncCluster] = useAsync(() =>
+    retryWithRelogin(appCtx, clusterUri, () =>
+      appCtx.clustersService.syncRootCluster(
+        routing.ensureRootClusterUri(clusterUri)
+      )
+    )
+  );
 
   return (
     <Document visible={props.visible}>
-      <ClusterState
-        clusterUri={clusterUri}
-        rootCluster={rootCluster}
-        cluster={cluster}
-        onLogin={logIn}
-      />
+      <Layout>
+        <ClusterState
+          clusterName={clusterName}
+          clusterUri={clusterUri}
+          rootCluster={rootCluster}
+          cluster={cluster}
+          syncCluster={syncCluster}
+          clusterSyncAttempt={clusterSyncAttempt}
+        />
+      </Layout>
     </Document>
   );
 }
 
 function ClusterState(props: {
   clusterUri: uri.ClusterUri;
+  clusterName: string;
   rootCluster: Cluster;
   cluster: Cluster | undefined;
-  onLogin(): void;
+  syncCluster(): void;
+  clusterSyncAttempt: Attempt<void>;
 }) {
   if (!props.rootCluster.connected) {
     return (
-      <RequiresLogin clusterUri={props.clusterUri} onLogin={props.onLogin} />
+      <RequiresLogin
```

Comment:
This scenario won't even happen, right? AFAIK it's now impossible for `rootCluster.connected` to change its state while the app is running. I wonder if we should just delete this conditional and state.

Reply:
We set `cluster.connected = false` in `logout()` https://github.com/gravitational/teleport/blob/7ce3d7a102b59bfbce950a83348beffa2254ed42/web/packages/teleterm/src/ui/services/clusters/clustersService.ts#L92 but just after that we change the workspace. So in theory, we have the intermediate state when the root cluster is not connected. This makes me unsure if we should remove it 🤔

Reply:
Ah, bless TypeScript. Let's keep it then. IIRC this was used as an actual view that was displayed to the user but since then we made changes to the whole app initialization process where it wasn't actively used.